RecoMuon/MuonSeedGenerator/src/MuonSeedOrcaPatternRecognition.cc

```diff
@@ -222,6 +274,22 @@ void MuonSeedOrcaPatternRecognition::produce(const edm::Event& event, const edm:
     tmp = filterSegments(muonMeasurements->recHits(ME11Bwd,event), endcapdThetaCut);
     copy(tmp.begin(),tmp.end(),back_inserter(all));
 
+    if (gemBackwardLayers.size()){
+      if (gemBackwardLayers.size() > 2){
+	const DetLayer* GE21Bwd = gemBackwardLayers[3];
+	tmp = filterSegments(muonMeasurements->recHits(GE21Bwd,event), endcapdThetaCut);
+	copy(tmp.begin(),tmp.end(),back_inserter(all));
```

Comment:
I think it would be best to add an appropriate reserve before using `back_inserter` (in all instances in this file):
```
all.reserve(all.size()+tmp.size())
```